components/templates/src/filters.rs

```diff
@@ -201,4 +223,44 @@ mod tests {
             assert_eq!(result.unwrap(), to_value(expected).unwrap());
         }
     }
+
+    #[test]
+    fn num_format_filter() {
+        let tests = vec![
+            (100, "100"),
+            (1_000, "1,000"),
+            (10_000, "10,000"),
+            (100_000, "100,000"),
+            (1_000_000, "1,000,000"),
+        ];
+
+        for (input, expected) in tests {
+            let args = HashMap::new();
+            let result = num_format(&to_value(input).unwrap(), &args);
+            let result = dbg!(result);
+            assert!(result.is_ok());
+            assert_eq!(result.unwrap(), to_value(expected).unwrap());
+        }
+    }
+
+    #[test]
+    fn num_format_filter_with_locale() {
+        let tests = vec![
+            ("en", 1_000_000, "1,000,000"),
+            ("en-IN", 1_000_000, "10,00,000"),
+            // Note:
+            // U+202F is the "NARROW NO-BREAK SPACE" code point.
+            // When displayed to the screen, it looks like a space.
+            ("fr", 1_000_000, "1\u{202f}000\u{202f}000"),
```

Comment:
Why not just use a space?

Reply:
@Keats This is coming from `num_format`. I think it's so you don't have numbers wrapping across line boundaries when they're split.

Reply:
Apart from "it doesn't line break" it's also narrower than a normal inter-word space. These are typographical details which in sum will give the published blog / website a more professional look. And since the Unicode character is standardized, all user agent can be expected to deal with it (even things like TTS).